constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/Violation.java

```diff
@@ -1,28 +1,36 @@
 package gov.nasa.jpl.aerie.constraints.model;
 
+import gov.nasa.jpl.aerie.constraints.time.Interval;
 import gov.nasa.jpl.aerie.constraints.time.Windows;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.StreamSupport;
 
 public final class Violation {
   public final List<Long> activityInstanceIds;
   public final List<String> resourceNames;
-  public final Windows violationWindows;
+  public final List<Interval> violationWindows;
 
   public Violation(final List<Long> activityInstanceIds, final List<String> resourceNames, final Windows violationWindows) {
     this.activityInstanceIds = new ArrayList<>(activityInstanceIds);
     this.resourceNames = new ArrayList<>(resourceNames);
-    this.violationWindows = violationWindows;
+    this.violationWindows = StreamSupport.stream(violationWindows.spliterateEqualTo(true), false).toList();
   }
 
   public Violation(final Windows violationWindows) {
     this(new ArrayList<>(), new ArrayList<>(), violationWindows);
   }
 
+  public Violation(final List<Long> activityInstanceIds, final List<String> resourceNames, final List<Interval> intervals) {
+    this.activityInstanceIds = activityInstanceIds;
+    this.resourceNames = resourceNames;
+    this.violationWindows = intervals;
+  }
+
   public Violation clone() {
-    return new Violation(List.copyOf(this.activityInstanceIds), List.copyOf(this.resourceNames), new Windows(violationWindows));
+    return new Violation(new ArrayList<>(this.activityInstanceIds), new ArrayList<>(this.resourceNames), new ArrayList<>(violationWindows));
```

Comment:
Yes, it was because it needed to be mutable. I'm almost done refactoring to use non-mutating methods, so the need has probably gone away now.

Reply:
Nope, its still a problem. Honestly I'm not sure what is different about `Violation` that now requires mutability. ASTTests for `ForEachActivity` fail when calling `addActivityId` because of immutability, which is baffling because that is unchanged and should have had the same error before this PR. I'll copy the other two lists, but not `activityInstanceIds`.